server/app/controllers/fusor/api/v21/deployments_controller.rb

```diff
@@ -81,11 +87,21 @@ def redeploy
     end
 
     def validate
+      error_syncing_openstack = nil
+      begin
+        sync_openstack
+      rescue => e
+        error_syncing_openstack =  _("Error contacting Openstack undercloud #{e.message}")
+      end
+
```

Comment:
so it does always get called, but line 213 checks deploy_openstack? before attempting to sync.